### ISSUE TITLE
How to move the search bar higher?

### ISSUE THREAD

**Issue by patrik-martinko** [Flow-Launcher/Flow.Launcher#1140]
Is there any chance to move the search bar higher? When I use Flow Launcher on Surface Pro in the landscape mode, the search results are hidden behind the touch keyboard.

![image](https://user-images.githubusercontent.com/12855743/164744141-d398217b-b0ee-463e-94a3-1a61823c3ebc.png)

I tried to change the `WindowTop` setting in the `Settings.json` file, but it didn't help.

**Comment by taooceros** [comment]
You can check the option of remembering last opened position in the setting and drag the window higher.

**Comment by patrik-martinko** [comment]
Yes, I know this option but then the search bar is not horizontally centered when I connect/disconnect from the monitor, rotate the tablet from/to portrait/landscape mode so it is even worse 😕

![image](https://user-images.githubusercontent.com/12855743/164891376-a97a7314-70e0-4a73-a14b-522e4568f98a.png)
![image](https://user-images.githubusercontent.com/12855743/164891381-d9ec13f3-89b9-4320-afb4-296f304e8b8b.png)

**Comment by taooceros** [comment]
Ah I see your issue. That probably requires a bit change in the setting option, and the centered algorithm is not depending on the theme file.
You can manually adjust part of the code here to hard code the search bar higher, but that probably is not something easy to be done in general for now. Sorry
https://github.com/Flow-Launcher/Flow.Launcher/blob/15fd62a8b0cedbc181ab60493e8f7d9cf1d2a2d8/Flow.Launcher/MainWindow.xaml.cs#L183-L195

**Comment by patrik-martinko** [comment]
Thanks. This would help but I should always change the code (and probably build the app?) with each update so this solution isn't not very convenient 😕.

Better solution would be to have a setting for example "Window Horizontal Position" with a few options.

![Screenshot 2022-04-23 191558](https://user-images.githubusercontent.com/12855743/164933786-d9809569-3096-40ad-8028-2ce714967e99.png)

I think that many people use touch devices today.

**Comment by taooceros** [comment]
Yeah I agree. @onesounds what do you think about implementing this.

**Comment by jjw24** [comment]
When remember last launch is not selected, the bar will re-centers will itself in tablet mode on next appear, so this is a request to add an height option on top so it will be center position + user defined height right?

**Comment by patrik-martinko** [comment]
> When remember last launch position is not selected, the bar will re-centers will itself in tablet mode on next appear, so this is a request to add an height option on top so it will be center position + user defined height right?

Correct.

**Comment by onesounds** [comment]
I agree that additional settings for the window position are needed. I've tried it before, but there were many things that needed to be fixed.
> Yeah I agree. @onesounds what do you think about implementing this.

**Comment by enoversum** [comment]
I'm also voting for this option. I use the portable version of this app and use it in different screen scenarios (one 4K screen vs. two 1080p screens), and unless I untick to save the window position, it's always at the top. Some kind of vertical position + centred would surely help, even best if it could be set in %. I understand it's probably not the most sought-after feature, and probably not concerning too many people.

**Comment by onesounds** [comment]
![image](https://user-images.githubusercontent.com/6903107/189780795-6a25a997-88da-4a6c-8769-bc84a0ffbb2a.png)

https://github.com/Flow-Launcher/Flow.Launcher/pull/1367

I don't think it's enough, but I'm going to add a setting that always opens at a specific location. I think it will be helpful when rotating the screen. Please use it later and give me your opinion.
